### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by rnestler**
See https://github.com/tokio-rs/axum/blob/main/examples/testing/src/main.rs

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -139,3 +143,33 @@ async fn accept_form(
...
+    async fn test_index() {
+        let app = app(default_args());
+
+        let response = app
+            .oneshot(Request::builder().uri("/").body(Body::empty()).unwrap())
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+        assert_eq!(response.headers().get(CONTENT_TYPE).unwrap(), "text/html; charset=utf-8");
```
Comment (rnestler) [inline src/main.rs]: I'm not sure on what level we want to test, but I think asserting for the status code and the content type is already enough.

File: src/main.rs [hunk 2]
```diff
@@ -139,3 +143,93 @@ async fn accept_form(
...
+                    .uri("/")
+                    .header(CONTENT_TYPE, "multipart/form-data;boundary=95685543938383789682253523760123")
+                    .body(Body::from(
+                            "--95685543938383789682253523760123\r\n\
+                            Content-Disposition: form-data; name=\"file\"; filename=\"test.txt\"\r\n\
+                            Content-Type: text/plain\r\n\
+                            \r\n\
+                            hello\
+                            \r\n\
+                            --95685543938383789682253523760123--"
```
Comment (rnestler) [inline src/main.rs:224]: During development of this I noticed that our error-handling is quite bare-bones with all these unwraps :wink: 

Can also be reproduced with `httpie/curl`:
```
http --multipart localhost:3001 file='foo'

http: LogLevel.ERROR: ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response')) while doing a POST request to URL: http://localhost:3001/
```

File: src/main.rs [hunk 3]
```diff
@@ -139,3 +143,96 @@ async fn accept_form(
...
     show_index(Some(&msg))
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use axum::body::Body;
+    use axum::http::header::CONTENT_TYPE;
+    use axum::http::{Request, StatusCode};
```
Comment (dbrgn) [inline src/main.rs]: Nothing important, but I'd generally merge imports per crate.

**Review by dbrgn — APPROVED** [review]
LGTM!

File: src/main.rs (hunk 3 above)
Comment (rnestler) [reply src/main.rs]: Fixed.

**Pull request merged**